restcomm/restcomm.testsuite/src/test/java/org/restcomm/connect/testsuite/http/CacheServletTest.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * TeleStax, Open Source Cloud Communications
+ * Copyright 2011-2014, Telestax Inc and individual contributors
+ * by the @authors tag.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation; either version 3 of
+ * the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+package org.restcomm.connect.testsuite.http;
+
+import com.sun.jersey.api.client.Client;
+import com.sun.jersey.api.client.ClientResponse;
+import com.sun.jersey.api.client.WebResource;
+import junit.framework.Assert;
+import org.apache.log4j.Logger;
+import org.jboss.arquillian.container.test.api.Deployer;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.junit.Arquillian;
+import org.jboss.arquillian.test.api.ArquillianResource;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.jboss.shrinkwrap.resolver.api.maven.Maven;
+import org.junit.Test;
+import org.junit.FixMethodOrder;
+import org.junit.runners.MethodSorters;
+import org.junit.runner.RunWith;
+import org.restcomm.connect.commons.Version;
+
+import java.net.URL;
+
+/**
+ * Test the servlet serving the media cache content is working.
+ *
+ * The restcomm configuration will be filtered by maven to replace the cache path
+ * to be in target/test-classes
+ *
+ */
+@RunWith(Arquillian.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class CacheServletTest extends EndpointTest {
+
+    private final static Logger logger = Logger.getLogger(CacheServletTest.class.getName());
+
+    private static final String version = Version.getVersion();
+
+    @ArquillianResource
+    private Deployer deployer;
+    @ArquillianResource
+    URL deploymentUrl;
+
+
+    @Test
+    public void testMediaDownloading() {
+
+        Client jersey = Client.create();
+        WebResource resource = jersey.resource(getResourceUrl("/cache/sample.wav"));
```

Comment:
```
RQNT 147483687 mobicents/ivr/4@10.0.0.28:2427 MGCP 1.0
N:restcomm@10.0.0.28:2727
R:AU/oc(N),AU/of(N)
X:8
S:AU/pa(an=https://staging.restcomm.com:443/restcomm/cache/ACae6e420f425248d6a26948c17a9e2acf/101cc31d1b07e8285c40d13d83f63da9caaaf21155a321f231980601bac555de.wav it=1)
```
@jaimecasero i am check current command sent by RC to RMS for playing a file
i notice `/ACae6e420f425248d6a26948c17a9e2acf/`
i dont think we should have account sids in url but sharing this to see the impact (if any)

Reply:
https://github.com/RestComm/Restcomm-Connect/blob/master/restcomm/restcomm.interpreter/src/main/java/org/restcomm/connect/interpreter/BaseVoiceInterpreter.java#L565

Reply:
well how we compose URL is not really a concern of this servlet. This servlet only concern is serve/expose  the content outside RC, so RMS may consume it. 

Having said that,we can review the URL composition model in the cache. WE surely need to phisically distribute the files in the dir, we dont want all files to be in same dir, we could hit a OS limit of files in dir soon.